src/weights.jl

```diff
@@ -9,20 +10,21 @@ and stores the `values` (`V<:RealVector`) and `sum` (`S<:Real`).
 """
 macro weights(name)
     return quote
-        mutable struct $name{S<:Real, T<:Real, V<:AbstractVector{T}} <: AbstractWeights{S, T, V}
-            values::V
+        mutable struct $name{S<:Real, T<:Real} <: AbstractWeights{S, T}
+            values::AbstractVector{T}
             sum::S
         end
         $(esc(name))(vs) = $(esc(name))(vs, sum(vs))
     end
 end
 
 length(wv::AbstractWeights) = length(wv.values)
-values(wv::AbstractWeights) = wv.values
 sum(wv::AbstractWeights) = wv.sum
 isempty(wv::AbstractWeights) = isempty(wv.values)
 size(wv::AbstractWeights) = size(wv.values)
 
+convert(::Type{Vector}, wv::AbstractWeights) = wv.values
```

Comment:
```suggestion
convert(::Type{AbstractVector}, wv::AbstractWeights) = wv.values
convert(::Type{Vector}, wv::AbstractWeights) = convert(Vector, wv.values)
```

(Probably also need to add a test to cover both methods.)